ignite/handlers/param_scheduler.py

```diff
@@ -16,6 +17,305 @@
+    """Update a parameter during training by using a step function.
+    This function decays the parameter value by gamma every step_size.
+    Based on StepLR from Pytorch.
+    https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L377
+
+    Args:
+        parameter_setter: function that sets the required parameter value.
+        initial_value: Starting value of the parameter.
+        gamma: Multiplicative factor of parameter value decay.
+        step_size: Period of parameter value decay.
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+    .. versionadded:: 0.6.0
+    """
+
+    def __init__(
+        self,
+        parameter_setter: Callable,
+        initial_value: float,
+        gamma: float,
+        step_size: int,
+        param_name: str,
+        save_history: bool = False,
+    ):
+        super(StepAnyParameterScheduler, self).__init__(parameter_setter, param_name, save_history)
+        self.initial_value = initial_value
+        self.gamma = gamma
+        self.step_size = step_size
+        self._state_attrs += ["initial_value", "gamma", "step_size"]
+
+    def get_param(self) -> Union[List[float], float]:
+        return self.initial_value * self.gamma ** (self.event_index // self.step_size)
+
+
+class MultiStepAnyParameterScheduler(AnyParameterScheduler):
+    """Update a parameter during training by using a multi step function.
+    The function decays the parameter value by gamma once the number of steps reaches one of the milestones.
+    Based on MultiStepLR from Pytorch.
+    https://github.com/pytorch/pytorch/blob/master/torch/optim/lr_scheduler.py#L424
+
+    Args:
+        parameter_setter: function that sets the required parameter value.
+        initial_value: Starting value of the parameter.
+        gamma: Multiplicative factor of parameter value decay.
+        milestones: List of step indices. Must be increasing.
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+    .. versionadded:: 0.6.0
+    """
+
+    def __init__(
+        self,
+        parameter_setter: Callable,
+        initial_value: float,
+        gamma: float,
+        milestones: List[int],
+        param_name: str,
+        save_history: bool = False,
+    ):
+        super(MultiStepAnyParameterScheduler, self).__init__(parameter_setter, param_name, save_history)
+        self.initial_value = initial_value
+        self.gamma = gamma
+        self.milestones = milestones
+        self._state_attrs += ["initial_value", "gamma", "milestones"]
+
+    def get_param(self) -> Union[List[float], float]:
+        return self.initial_value * self.gamma ** bisect_right(self.milestones, self.event_index)
+
+
+class OptimizerParamScheduler(ParamScheduler):
```

Comment:
@fco-dv this is BC-breaking change, I think we should keep BC compatibilty

Reply:
Ok I see, do you think we should keep `ParamScheduler` as it is and override the methods in `StateParameterScheduler(ParamScheduler)`?